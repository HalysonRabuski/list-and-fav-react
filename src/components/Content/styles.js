import styled from "styled-components";
import { Layout } from "antd";
const { Content } = Layout;

export const StyledContent = styled(Content)`
  &.ant-layout-content {
    padding: 20px;
  }
`;
