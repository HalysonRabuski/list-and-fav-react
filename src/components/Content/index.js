import React from "react";
import { StyledContent } from "./styles.js";

function Content(props) {
  return (
    <>
      <StyledContent data-testid={`content`}>{props.children}</StyledContent>
    </>
  );
}

export default Content;
