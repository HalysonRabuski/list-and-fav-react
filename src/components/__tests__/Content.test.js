import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Content from "../Content";

afterEach(() => {
  cleanup();
});

test("should render content component", () => {
  render(
    <Content>
      <p>test</p>
    </Content>
  );
  const contentElement = screen.getByTestId("content");
  expect(contentElement).toBeInTheDocument();
  expect(contentElement).toHaveTextContent("test");
});

test("should render content component with its children", () => {
  const contentElement = renderer
    .create(
      <Content>
        <p>test</p>
      </Content>
    )
    .toJSON();
  expect(contentElement).toMatchSnapshot();
});
