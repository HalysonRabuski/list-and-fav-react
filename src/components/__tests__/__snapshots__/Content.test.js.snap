// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render content component with its children 1`] = `
<main
  className="ant-layout-content sc-bdvvaa gombUE"
  data-testid="content"
>
  <p>
    test
  </p>
</main>
`;
